sdk/pinocchio/src/entrypoint/mod.rs

```diff
@@ -112,14 +107,16 @@ const STATIC_ACCOUNT_DATA: usize = core::mem::size_of::<Account>() + MAX_PERMITT
 /// handler manually.
 #[macro_export]
 macro_rules! entrypoint {
-    ( $process_instruction:ident ) => {
-        entrypoint!($process_instruction, { $crate::MAX_TX_ACCOUNTS });
-    };
-    ( $process_instruction:ident, $maximum:expr ) => {
-        $crate::program_entrypoint!($process_instruction, $maximum);
+    ( $process_instruction:expr ) => {
```

Comment:
Changing `ident` to `expr` (in all entrypoint macros) allows the use of any function pointer not just identifiers in current scope.

Reply:
Good point, I will add some comments there.